windowsforms/FlexGridShowcaseDemo/cs/CustomRowDetail.cs

```diff
@@ -0,0 +1,64 @@
+﻿using C1.Win.FlexGrid;
+using System;
+using System.Data;
+using System.Drawing;
+using System.Linq;
+using System.Windows.Forms;
+
+namespace FlexGridShowcaseDemo
+{
+    /// <summary>
+    /// Custom row detail class which shows label with notes about employee.
+    /// </summary>
+    public class CustomRowDetail :  Label, IC1FlexGridRowDetail
+    {
+
+        /// <summary>
+        /// Used to setup control before showing of it.
+        /// </summary>
+        /// <param name="parentGrid">FlexGrid which displays detail control.</param>
+        /// <param name="rowIndex">Index of parent detail row.</param>
+        void IC1FlexGridRowDetail.Setup(C1FlexGrid parentGrid, int rowIndex)
+        {
+            var bs = new BindingSource(parentGrid.DataSource as DataSet, "Products");
+            bs.Position = parentGrid.Rows[rowIndex].DataIndex;
+            var row = bs.Current as DataRowView;
+
+            // Formatting text
+            var title = (from s in row.Row.Table.Columns.Cast<DataColumn>() select s)
+                .Where(x => x.ColumnName != "Name")
+                .Select(x => x.ColumnName);
+
+            var details = title
+                .Select(x => string.Format("{0}: {1}", x, row[x]))
+                .ToArray();
```

Comment:
No need to allocate an array, [`string.Join` takes `IEnumerable`](https://docs.microsoft.com/en-us/dotnet/api/system.string.join?view=netcore-3.1#System_String_Join_System_String_System_Collections_Generic_IEnumerable_System_String__)

Reply:
Fixed with a74c3d1477489f5e60c764dd5aab171602a117f7.